server/plugin.go

```diff
@@ -399,6 +399,18 @@ func (p *Plugin) AddAutolinks(projectList jira.ProjectList, baseURL string) erro
 	}
 
 	client := autolinkclient.NewClientPlugin(p.API)
+	
+	var keys []string
+	for _,autolink:= range installList {
+		keys = append(keys,autolink.Name)
+	}
+
+	// Deleting the old autolinks if already present
+	if err := client.Delete(keys...); err != nil {
+		return fmt.Errorf("unable to delete autolinks: %w", err)
+	}
```

Comment:
@avas27JTG This will likely have issues in HA environments, with multiple instances of the plugin running at the same time. They will be fighting over reading/deleting/creating the autolinks. @hanzei What are your thoughts on this?

Reply:
Can we simply have a giant lock around the function and check first, if the autolinks need to get generated?